pkg/ipamd/datastore/data_store.go

```diff
@@ -207,13 +228,13 @@ func (p *ENIPool) AssignedIPv4Addresses() int {
 }
 
 // FindAddressForSandbox returns ENI and AddressInfo or (nil, nil) if not found
-func (p *ENIPool) FindAddressForSandbox(ipamKey IPAMKey) (*ENI, *AddressInfo) {
+func (p *ENIPool) FindAddressForSandbox(ipamKey IPAMKey) (*ENI, *AssignedIPv4Addresses, *AddressInfo) {
```

Comment:
nit: map is passed by value, i don't think we need pointer receiver here.

Reply:
There are quite a places even in the legacy code where we are passing the pointer (`AddressInfo*`). It would need quite some code changes so will discuss with you and take it up in a followup PR.

Reply:
Discussed with Yang, there is lot of legacy code which needs this clean up, so will take up in future PRs.